federatedscope/llm/trainer/trainer.py

```diff
@@ -36,9 +63,21 @@ def _hook_on_batch_forward(self, ctx):
         ctx.loss_batch = CtxVar(loss, LIFECYCLE.BATCH)
         ctx.batch_size = CtxVar(len(labels), LIFECYCLE.BATCH)
 
+    def _hook_on_batch_backward(self, ctx):
+        ctx.optimizer.zero_grad()
+        if self.use_accelerator:
```

Comment:
Current version is OK for non-accumulated gradients scenarios. While we should keep in mind that we need properly use `accelerator.accumulate(model)`, which can be a TODO item, see more details [here](https://huggingface.co/docs/accelerate/concept_guides/gradient_synchronization)